File: utils/test/random_test.py
import numpy as np

values = []

"""
np.random.seed(1)
for _ in range(10000000):
    values.append(np.random.randint(0,4))

print('{:.13f}'.format(np.var(values)))
"""

classes=[]
np.random.seed(1)
for _ in range(10000000):
    values.append(np.random.uniform())
    classes.append(int(values[-1] * 10)/10)

print('{:.13f}'.format(np.var(values)))

from collections import Counter
c = Counter(classes)
for n,v in c.most_common():
    print(n,v)

File: utils/test/random_test.lua
require 'torch'

local count = torch.FloatTensor(10000000)

--[[
torch.manualSeed(1)
for i=1, 10000000 do
    v = torch.random(1,4)
    count[i] = v
end

print(torch.var(count))
]]

local classes={}
torch.manualSeed(1)
for i=1, 10000000 do
    v = torch.uniform()
    count[i] = v
    idx = string.format('%d',v*10)
    num = classes[idx]
    if num then
        classes[idx]  = num + 1
    else
        classes[idx] = 1
    end
end

print(torch.var(count))
for i=0, 9 do
    print(i, classes[string.format('%d',i)])
end
